models/points.go

```diff
@@ -25,8 +25,20 @@ var (
 		' ': []byte(`\ `),
 		'=': []byte(`\=`),
 	}
+
+	// used to track length of binary-encoded time.Time
+	timeLen int
 )
 
+func init() {
+	binTime, err := time.Now().MarshalBinary()
+	if err != nil {
+		// we'll be marshaling time objects a lot
+		panic(err)
+	}
+	timeLen = len(binTime)
```

Comment:
Further down, in `(*point).MarshalBinary()`, we binary-encode the `time.Time` objects. And rather than guessing, or making assumptions that the encoded size will never change, I added this to figure out the encoded size on startup, then use that to allocate the required space later.